lib/web/fetch/index.js

```diff
@@ -120,130 +122,146 @@ class Fetch extends EE {
   }
 }
 
+// subscribersCheck will be called at the beginning of the fetch call
+// and will check if we have subscribers
+function subscribersCheck () {
+  return channels.tracingChannel.start.hasSubscribers ||
+  channels.tracingChannel.end.hasSubscribers ||
+  channels.tracingChannel.asyncStart.hasSubscribers ||
+  channels.tracingChannel.asyncEnd.hasSubscribers ||
+  channels.tracingChannel.error.hasSubscribers
+}
+
 // https://fetch.spec.whatwg.org/#fetch-method
-function fetch (input, init = undefined) {
-  webidl.argumentLengthCheck(arguments, 1, { header: 'globalThis.fetch' })
+function fetch (input, init = {}) {
+  const hasSubscribers = subscribersCheck()
+  let context
+  if (hasSubscribers) context = { input, init, result: null, error: null }
+  return channels.tracingChannel.tracePromise(async () => {
```

Comment:
`tracePromise` should almost _never_ be used give the extra overhead it adds.

Reply:
With the recent early exit it should be nothing more than the following, with no subscribers:

```js
tracePromise(fn, thisArg, ...args) {
  if (!this.hasSubscribers) {
    return Reflect.apply(fn, thisArg, ...args)
  }
  // ...
}
```

As for the closure, that _is_ why `tracePromise` passes through `thisArg` and `...args`--so that it can be used to replace a function call where possible, rather than executing a closure.

In _most_ cases it should be generally fine to use, but the pattern it implements is specifically documented to enable the particularly performance-sensitive cases to avoid a function wrap altogether, where absolutely necessary.

Reply:
The promise I'm referring to is the one of the closure.